libraries/matrix/impl/src/main/kotlin/io/element/android/libraries/matrix/impl/auth/RustMatrixAuthenticationService.kt

```diff
@@ -71,6 +73,14 @@ class RustMatrixAuthenticationService @Inject constructor(
         return sessionStore.isLoggedIn()
     }
 
+    override fun incrementCacheIdx() {
```

Comment:
These APIs feel opaque.
It is unclear what's the intent behind them.
The name "cache index" is misleading because it leads the reader into thinking this API has something to do with the cache.
Instead it seems it's only used to call `matrixClientsHolder.removeAll()` every time the cache is cleared.
If we could call `removeAll()` from the `ClearCacheUseCase` directly we wouldn't need these.

Reply:
The main role is to trigger a reload of the app, where the Flow is used. But I admit I can rework this part.